documents/coredocument.go

```diff
@@ -177,6 +184,7 @@ func (cd *CoreDocument) PrepareNewVersion(collaborators []string, initSalts bool
 		Roles:              cd.Document.Roles,
 		ReadRules:          cd.Document.ReadRules,
 		Nfts:               cd.Document.Nfts,
+		SignatureData:      new(coredocumentpb.SignatureData),
```

Comment:
We don't need this if we have nil check for `AppendSignatures`, right?

Reply:
there are couple other places where we access signature data before it has been initialized.